File: backend/src/main/java/com/ecommerce/ecommerce/business/abstracts/UserService.java
package com.ecommerce.ecommerce.business.abstracts;

import java.util.List;

import com.ecommerce.ecommerce.core.entities.User;
import com.ecommerce.ecommerce.core.utilities.results.Result;
import com.ecommerce.ecommerce.core.utilities.results.dataResults.DataResult;

public interface UserService {

    DataResult<List<User>> getAll();

    DataResult<User> getById(int id);

    DataResult<User> getByEmail(String email);

    DataResult<User> add(User user);

    DataResult<User> update(User user);

    Result delete(int id);
}

File: frontend/src/app/services/user.service.ts
import { SingleResponseModel } from './../models/singleResponseModel';
import { User } from './../models/user';
import { Observable } from 'rxjs';
import { Constants } from './../constants/constants';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ListResponseModel } from '../models/listResponseModel';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  apiUrl: string = Constants.apiUrl + "users/";

  constructor(private httpClient: HttpClient) { }

  getUsers(): Observable<ListResponseModel<User>> {
    return this.httpClient.get<ListResponseModel<User>>(this.apiUrl + "getAll");
  }

  getUserByEmail(email: string): Observable<SingleResponseModel<User>> {
    return this.httpClient.get<SingleResponseModel<User>>(this.apiUrl + "getByEmail?email=" + email);
  }

  getUserById(userId: number): Observable<SingleResponseModel<User>> {
    return this.httpClient.get<SingleResponseModel<User>>(this.apiUrl + "getById?id=" + userId);
  }

  deleteUser(userId: number): Observable<SingleResponseModel<User>> {
    return this.httpClient.post<SingleResponseModel<User>>(this.apiUrl + "delete?id=" + userId, null);
  }

  updateUser(user: User): Observable<SingleResponseModel<User>> {
    return this.httpClient.post<SingleResponseModel<User>>(this.apiUrl + "update", user);
  }
}
